### ISSUE TITLE
golang kubernetes provider is erroneously rendering to yaml directory if renderToYamlDirectory is unset

### ISSUE THREAD

**Issue by jsravn** [pulumi/pulumi-kubernetes#1684]
## Steps to reproduce

Try to create a kubernetes provider and create a resource with it:
```
	kubeconf := createKubeConfig(gkeCluster)
	clusterProvider, err := kubernetes.NewProvider(d.ctx, "gke-cluster", &kubernetes.ProviderArgs{
		Kubeconfig: kubeconf,
	})
	if err != nil {
		return err
	}

	ns, err := corev1.NewNamespace(d.ctx, "kube-ready", &corev1.NamespaceArgs{
		Metadata:   metav1.ObjectMetaArgs{
			Name:   pulumi.String("kube-ready"),
		},
	}, pulumi.Provider(clusterProvider))
	if err != nil {
		return err
	}
// and so on
```

When I run `pulumi up`, instead of applying to the cluster, it creates yaml files in `null/1-manifest` folder. The problem is the RenderToYamlDirectory check is not considering a nil value as unset. As a workaround, I can set `RenderToYamlDirectory` to `""` which fixes it.

On pulumi v3.10.3.

Expected: kubernetes provider works by just setting the kubeconfig field
Actual: it erroneously renders to yaml in the local filesystem

**Comment by clstokes** [comment]
@jsravn can you share your `go.mod`/`go.sum` too?

**Comment by lblackstone** [comment]
This is fixed by https://github.com/pulumi/pulumi-kubernetes/pull/1678 which is included in the v3.6.1 release
